Aymen-Khanfir/ets
# feat: add Husky hooks for commit linting

### Summary
This PR introduces Husky hooks for commit linting, sets up lint-staged for pre-commit validations, and configures ESLint and Prettier.

### Changes
**Changes include:**
- Added Husky hooks (`pre-commit` and `commit-msg`).
- Configured commitlint to enforce conventional commit messages.
- Integrated lint-staged for pre-commit code formatting and linting.
- Set up ESLint and Prettier for code consistency.

### Why This Change Is Necessary
These changes are necessary to ensure consistent code quality, enforce commit message conventions, and automate formatting/linting tasks during development.

### Testing Instructions:
**Steps to test:**
1. Clone the repository and switch to the `feat/add-husky-hooks` branch.
2. Try committing a non-conventional commit message (it should fail).
3. Try committing a file with lint errors (it should fail and suggest fixes).
4. Ensure `pnpm prepare` installs Husky hooks successfully.

### Checklist
- [x] Husky hooks added.
- [x] Lint-staged configured.
- [x] ESLint and Prettier set up.
